tests/aws/services/events/test_event_patterns.py

```diff
@@ -0,0 +1,160 @@
+                absolute_filepath = os.path.join(root, filename)
+                files.append(absolute_filepath)
+
+    return files
+
+
+request_template_tuples = load_request_templates(REQUEST_TEMPLATE_DIR)
+
+SKIP_LABELS = [
+    # Failing exception tests:
+    "arrays_empty_EXC",
+    "content_numeric_EXC",
+    "content_numeric_operatorcasing_EXC",
+    "content_numeric_syntax_EXC",
+    "content_wildcard_complex_EXC",
+    "int_nolist_EXC",
+    "operator_case_sensitive_EXC",
+    "string_nolist_EXC",
+    # Failing tests:
+    "complex_or",
+    "content_anything_but_ignorecase",
+    "content_anything_but_ignorecase_list",
+    "content_anything_suffix",
+    "content_exists_false",
+    "content_ignorecase",
+    "content_ignorecase_NEG",
+    "content_ip_address",
+    "content_numeric_and",
+    "content_prefix_ignorecase",
+    "content_suffix",
+    "content_suffix_ignorecase",
+    "content_wildcard_nonrepeating",
+    "content_wildcard_repeating",
+    "content_wildcard_simplified",
+    "dot_joining_event",
+    "dot_joining_pattern",
+    "nested_json_NEG",
+    "or-exists",
+    "or-exists-parent",
+]
+
+
+# TODO: extend these test cases based on the open source docs + tests: https://github.com/aws/event-ruler
+#  For example, "JSON Array Matching", "And and Or Relationship among fields with Ruler", rule validation,
+#  and exception handling.
+@pytest.mark.skipif(is_v2_provider(), reason="V2 provider does not support this feature yet")
+@pytest.mark.parametrize(
+    "request_template,label", request_template_tuples, ids=[t[1] for t in request_template_tuples]
+)
+@markers.aws.validated
+def test_test_event_pattern(aws_client, snapshot, request_template, label):
+    """This parametrized test handles three outcomes:
+    a) MATCH (default): The EventPattern matches the Event yielding true as result.
+    b) NO MATCH (_NEG suffix): The EventPattern does NOT match the Event yielding false as result.
+    c) EXCEPTION (_EXC suffix): The EventPattern is invalid and raises an exception.
+    """
+    if label in SKIP_LABELS and not is_aws_cloud():
+        pytest.skip("Not yet implemented")
+
+    event = request_template["Event"]
+    event_pattern = request_template["EventPattern"]
+
+    if label.endswith("_EXC"):
+        with pytest.raises(Exception) as e:
+            aws_client.events.test_event_pattern(
+                Event=json.dumps(event),
+                EventPattern=json.dumps(event_pattern),
+            )
+        exception_info = {"exception_type": type(e.value), "exception_message": e.value.response}
+        snapshot.match(label, exception_info)
+    else:
+        response = aws_client.events.test_event_pattern(
+            Event=json.dumps(event),
+            EventPattern=json.dumps(event_pattern),
+        )
+
+        # Validate the test intention: The _NEG suffix indicates negative tests (i.e., a pattern not matching the event)
+        if label.endswith("_NEG"):
```

Comment:
It might be useful in the future to snapshot the result together with a hash of the request because currently, it's possible to modify the test case after being AWS-validated; kinda proper testing police ;)